### PR TITLE
H-2473, H-2584: Add new HASH app homepage screens

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

Adds new homepage screens (internal [designs](https://www.figma.com/design/gydVGka9FjNEg9E2STwhi2/HASH?node-id=11634-6408&t=9Kvph0gBWEHFrelI-4)) for:
- logged out users
- logged in users who are on the waitlist
- logged in users who have access

## 🚫 Blocked by

- [x] Merging [the PR](https://github.com/hashintel/internal/pull/152) (internal) which adds the 'get waitlist' functionality
  - technically the new screens can be added without this PR, they just won't show a waitlist position and will fall back to 'You are on the waitlist'
  
## 🔍 What does this change?

As well as adding the screens:
- Consolidates some of our modal styles where modals have a header and close button
- H-2584: Create an `@example` organization in hosted HASH (app.hash.ai, locally)

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

## 🛡 What tests cover this?

- Some existing

## ❓ How to test this?

1. Check out the homepage as a logged out user, logged in user
2. To test waitlist:
  - run the internal API off the branch linked above
  - in the HASH `.env.local` set an allow list:
```
USER_EMAIL_ALLOW_LIST=["alice@example.com", "bob@example.com", "admin@example.com"]
```
  - sign up with an email that is already on the HASH list (e.g. your own that is signed up to app.hash.ai). first request will take some time as the internal API needs to build its cache of the list
  - try the early access form

#### Logged out -> Sign up, on waitlist -> sign in to user with access

https://github.com/hashintel/hash/assets/37743469/bc99c557-97fc-4cc5-9485-ac44e2baa972

#### User not on allow list, self-hosted instance

Screenshot 2024-05-22 at 12 19 28

### CODE REVIEW

**Comment by vilkinsons** [comment]
Slacked a few comments over.

**Review by vilkinsons — APPROVED** [review]
Functional approval ✅👍

**Review by vilkinsons — APPROVED** [review]

File: apps/hash-api/src/graph/ensure-system-graph-is-initialized/migrate-ontology-types/migrations/017-add-use-case-form-and-example-org.migration.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        title: "Willing To Pay",
+        description: "The amount that someone is willing to pay for something",
+        possibleValues: [{ primitiveDataType: "text" }],
+      },
+      webShortname: "hash",
+      migrationState,
+    },
+  );
+
+  const _prospectiveUserDefinitionEntityType =
```
Comment (benwerner01) [inline apps/hash-api/src/graph/ensure-system-graph-is-initialized/migrate-ontology-types/migrations/017-add-use-case-form-and-example-org.migration.ts:93]: Just curious, why are we not attaching this information directly to the `User` entity? I assume we will need to use the `ownedById` of the entity to determine which account this "Prospective User" is associated with, so I suppose this is not a big deal.

File: apps/hash-frontend/src/pages/index.page/shared/typography.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    sx={[
+      ({ palette }) => ({
+        color: palette.gray[60],
+        fontSize: 24,
+        letterSpacing: "-2%",
+        fontWeight: 500,
+        mb: 3,
+        mt: 1,
+      }),
+      ...(Array.isArray(sx) ? sx : []),
```
Comment (benwerner01) [inline apps/hash-frontend/src/pages/index.page/shared/typography.tsx:42]:
```suggestion
      ...(Array.isArray(sx) ? sx : [sx]),
```

File: apps/hash-frontend/src/components/hooks/use-logout-flow.ts [hunk 3]
```diff
@@ -35,7 +35,7 @@ export const useLogoutFlow = () => {
            * hook on the current page.
            */
           await router.push({
-            pathname: "/signin",
+            pathname: "/",
             query: { return_to: router.asPath },
           });
```
Comment (benwerner01) [inline apps/hash-frontend/src/components/hooks/use-logout-flow.ts:40]: Not caused by this change, but I noticed an edge case when logging out with the sidebar collapsed where the contents overflow the screen width for some reason. I assume some state is not being updated correctly when the sidebar is collapsed or something related. 

https://github.com/hashintel/hash/assets/42802102/e81ea887-1c53-4024-af4c-258b487a9287

**Review by benwerner01 — COMMENTED** [review]
Thanks Ciaran, the changes look good and are working for me locally 👍 

Spotted one minor thing regarding a layout issue after logging out, but if it's not straightforward to address may not need to be addressed now as it's an edge case.

File: apps/hash-api/src/graph/ensure-system-graph-is-initialized/migrate-ontology-types/migrations/017-add-use-case-form-and-example-org.migration.ts (hunk 1 above)
Comment (CiaranMn) [reply apps/hash-api/src/graph/ensure-system-graph-is-initialized/migrate-ontology-types/migrations/017-add-use-case-form-and-example-org.migration.ts:93]: I figured these are temporary things and so didn't want to create further unnecessary types / links... we can always add them later if we want

File: apps/hash-frontend/src/components/hooks/use-logout-flow.ts (hunk 3 above)
Comment (CiaranMn) [reply apps/hash-frontend/src/components/hooks/use-logout-flow.ts:40]: Thanks for flagging, have fixed in https://github.com/hashintel/hash/pull/4502/commits/b7beeba5d7b2485068787603479ecbefc433ee69

**Review by vilkinsons — APPROVED** [review]

**Review by benwerner01 — APPROVED** [review]

**Pull request merged**